### DISCUSSION TITLE
ConfigureIIs is not recognized in v4.01

### DISCUSSION THREAD

**Discussion by mvollmer1** [wixtoolset/issues#7716] [Questions]
I have this configuration in my wxs file to run a  custom action if a condition is met...

<Custom Action="MS.SetConfigureIIs" Before="ConfigureIIs" 
        Condition="(&amp;ServiceFeature &lt; 3 AND &amp;ViewFeature &lt; 3) OR (!ServiceFeature &gt;= 3 AND !ViewFeature &gt;= 3)" />
  
The VS build returns this error:

Error	WIX0094	The identifier 'WixAction:InstallExecuteSequence/ConfigureIIs' could not be found. Ensure you have typed the reference correctly and that all the necessary inputs are provided to the linker.

Perhaps Wix does not recognize the ConfigureIIs built in action id?

I've checked the source for v3 & v4, the references for this are present in both...

source module: scasched.cpp

/********************************************************************
ConfigureIIs - CUSTOM ACTION ENTRY POINT for installing IIs settings

********************************************************************/
extern "C" UINT __stdcall ConfigureIIs(
    __in MSIHANDLE hInstall
    )

**Comment by mvollmer1** [comment]
`<Custom Action="MS.SetConfigureIIs" Before="ConfigureIIs" 
        Condition="(&amp;ServiceFeature &lt; 3 AND &amp;ViewFeature &lt; 3) OR (!ServiceFeature &gt;= 3 AND !ViewFeature &gt;= 3)" />
  `

**Comment by robmen** [comment]
`ConfigureIIs` isn't a built-in action. It comes from the `WixToolset.Iis.wixext`. You found the DLL entry point (good job!), but you're looking for the `Id` of the `CustomAction` element that calls that entry point. I won't spoil the learning experience. You're very close to connecting all the dots. 👍🏻

**Comment by luissagasta** [comment]
In v4, the name of the `ConfigureIIs` is `Wix4ConfigureIIs_X86` or `Wix4ConfigureIIs_X64`. You can use `$(sys.BUILDARCHSHORT)` for the architecture: `Wix4ConfigureIIs_$(sys.BUILDARCHSHORT)`
